### PR TITLE
Update text matrix for the 3 most recent versions

### LINKED ISSUE #1301: Update test matrix to new GitLab versions
### Feature Description

This issue will update the test matrix to the newly released versions of GitLab.

### Do you want to implement this?

- [X] I would like to implement this myself 👷

### PR DESCRIPTION
**Pull request by PatrickRice-KSC**
## Description

This PR just updates the test matrix to use the 3 most recent versions.

Closes #1301

### PR Checklist Acknowledgement

- [x] I acknowledge that all of the following items are true, where applicable:
  - Resource attributes match 1:1 the names and structure of the API resource in [the GitLab API documentation](https://docs.gitlab.com/ee/api/).
  - [Examples](https://github.com/gitlabhq/terraform-provider-gitlab/tree/main/examples) are updated with:
    - A \*.tf file for the resource/s with at least one usage example
    - A \*.sh file for the resource/s with an import example (if applicable)
  - New resources have at minimum a basic test with three steps:
    - Create the resource
    - Update the attributes
    - Import the resource
  - No new `//lintignore` comments were copied from existing code. (Linter rules are meant to be enforced on new code.)

### CODE REVIEW

**Comment by timofurrer** [comment]
most tests are failing because of https://github.com/gitlabhq/terraform-provider-gitlab/issues/1304

**Comment by PatrickRice-KSC** [comment]
Yeah, I think there are a couple that are new failures, but need to get the current ones solved before I can confirm that.

**Comment by timofurrer** [comment]
Could you please rebase onto the latest main (see #1304)?

**Comment by PatrickRice-KSC** [comment]
rebased - I think we're going to have some test failures just based on what I was seeing before, but we'll see here in a bit :)

**Comment by PatrickRice-KSC** [comment]
@timofurrer  | @armsnyder  | @Shocktrooper  - This is ready for review/merge. Apparently 100% of the failed tests were related to license issues, which is good; makes more sense 😄

**Comment by Shocktrooper** [comment]
@PatrickRice-KSC Security updates are out as of 2 days ago should we target those?
https://about.gitlab.com/releases/categories/releases/

**Comment by PatrickRice-KSC** [comment]
@Shocktrooper  - good catch, sure!

File: .github/workflows/pr-acceptance-ce.yml [hunk 1]
```diff
@@ -34,7 +34,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        gitlab-version: ["15.1.6-ce.0", "15.2.4-ce.0", "15.3.3-ce.0"]
+        gitlab-version: ["15.3.5-ce.0", "15.4.3-ce.0", "15.5.2-ce.0"]
```
Comment (Shocktrooper) [inline .github/workflows/pr-acceptance-ce.yml:37]:
```suggestion
        gitlab-version: ["15.3.5-ce.0", "15.4.4-ce.0", "15.5.2-ce.0"]
```

File: .github/workflows/push.yml [hunk 2]
```diff
@@ -97,7 +97,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        gitlab-version: ["15.1.6-ce.0", "15.2.4-ce.0", "15.3.3-ce.0"]
+        gitlab-version: ["15.3.5-ce.0", "15.4.3-ce.0", "15.5.2-ce.0"]
```
Comment (Shocktrooper) [inline .github/workflows/push.yml:100]:
```suggestion
        gitlab-version: ["15.3.5-ce.0", "15.4.4-ce.0", "15.5.2-ce.0"]
```

**Review by Shocktrooper — CHANGES_REQUESTED** [review]

**Review by timofurrer — APPROVED** [review]

**Pull request merged**